src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisWorkerService.java

```diff
@@ -178,10 +177,12 @@ private static String sanitizeParam(String param) {
                             }
                         }
                     })
-                    .doOnError(throwable -> {
+                    .onErrorResume(throwable -> {
                         if (!(throwable instanceof CancellationException)) {
                             LOGGER.error(throwable.toString(), throwable);
                         }
+                        stoppedPublisherService.publish(resultContext.getResultUuid(), resultContext.getRunContext().getReceiver());
```

Comment:
why do publish and delete here ? it's already done in the consumeCancel

Reply:
should this be in a separate PR ?

Reply:
Yes this should be inside the `if`
```
if (!(throwable instanceof CancellationException)) {
    LOGGER.error(throwable.toString(), throwable);
    stoppedPublisherService.publish(resultContext.getResultUuid(), resultContext.getRunContext().getReceiver());
    return resultRepository.delete(resultContext.getResultUuid()).then(Mono.empty());
}
return Mono.empty();

```

Reply:
it would be interesting to add a status in the stop message : 
status : OK or NOK
error message : a message